app/scripts/metamask-controller.js

```diff
@@ -87,6 +88,9 @@ module.exports = class MetamaskController extends EventEmitter {
     // the only thing that uses controller connections are open metamask UI instances
     this.activeControllerConnections = 0
 
+    // This is a map of message streams to various domains.
+    this.connections = {}
```

Comment:
I don't see another reference to this object. I think it's a great idea, by the way, and I am implementing something like it for the `accountsChanged` event using the permissions system.